src/DotNet.Status.Web/Controllers/GitHubHookController.cs

```diff
@@ -53,22 +59,187 @@ private async Task EnsureLabelsAsync()
+
+        private async Task<T> DeserializeGitHubWebHook<T>()
+        {
+            // Octokit has a serializer and types built for github rest apis, use them
+            string data;
+            using (var reader = new StreamReader(Request.Body))
+            {
+                data = await reader.ReadToEndAsync();
+            }
+
+            var payload = _serializer.Deserialize<T>(data);
+            return payload;
+        }
+
+        [GitHubWebHook(EventName = "pull_request_review_comment")]
+        public async Task<IActionResult> PullRequestReviewComment()
+        {
+            var payload = await DeserializeGitHubWebHook<PullRequestCommentPayloadWithChanges>();
+
+            string repo = payload.Repository.Owner.Login + "/" + payload.Repository.Name;
+            int number = payload.PullRequest.Number;
+            _logger.LogInformation("Received webhook for pull request {repo}#{number}", repo, number);
+
+            string title = payload.PullRequest.Title;
+            string uri = payload.Comment.HtmlUrl;
+            string username = payload.PullRequest.User.Login;
+            DateTimeOffset date = payload.PullRequest.UpdatedAt;
+            try
+            {
+                bool sent = false;
+                switch (payload.Action)
+                {
+                    case "created":
+                        sent = await _issueMentionForwarder.HandleIssueBody(null, payload.Comment.Body, title, uri, username, date);
+                        break;
+                    case "edited" when !string.IsNullOrEmpty(payload.Changes.Body?.From):
+                        sent = await _issueMentionForwarder.HandleIssueBody(payload.Changes.Body.From, payload.Comment.Body, title, uri, username, date);
+                        break;
+                }
+
+                if (sent)
+                {
+                    _logger.LogInformation("Sent teams notification for pull request {repo}#{number}", repo, number);
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogWarning(ex, "Unable to send teams notification for pull request {repo}#{number}", repo, number);
+            }
+
+            return Ok();
+        }
+
+        [GitHubWebHook(EventName = "issue_comment")]
+        public async Task<IActionResult> IssueComment()
+        {
+            var payload = await DeserializeGitHubWebHook<IssueCommentPayloadWithChanges>();
+
+            string repo = payload.Repository.Name;
+            int number = payload.Issue.Number;
+            _logger.LogInformation("Received comment webhook for issue {repo}#{number}", repo, number);
+
+            string title = payload.Issue.Title;
+            string uri = payload.Comment.HtmlUrl;
+            string username = payload.Issue.User.Login;
+            DateTimeOffset date = payload.Issue.UpdatedAt ?? DateTimeOffset.UtcNow;
```

Comment:
Should probably use a ISystemClock rather than pull now.  Though why wouldn't an issue have an updated time?

Reply:
I'm not really sure why it wouldn't. The property is nullable though. I'll use the system clock thing.